app/models/manageiq/providers/amazon/cloud_manager/metrics_capture.rb

```diff
@@ -133,7 +133,7 @@ def perf_capture_data_amazon(start_time, end_time)
     COUNTER_INFO.each do |i|
       timestamps = i[:amazon_counters].collect do |c|
         metrics_by_counter_name[c].keys unless metrics_by_counter_name[c].nil?
-      end.flatten.uniq.sort.compact
+      end.flatten.uniq.compact.sort
```

Comment:
@Fryguy ping me if you have reservations about the viability of this test. I'm totally open to expanding it if needed. That said what I did when I added the test was observe the .sort.compact failure.
```
     ArgumentError:
       comparison of Time with nil failed
     # ./app/models/manageiq/providers/amazon/cloud_manager/metrics_capture.rb:136:in `sort'
```
Then I updated metrics_capture.rb and reran the test and it passed.